commands/auth.ts

```diff
@@ -64,35 +57,52 @@ const ALLOWED_AUTH_METHODS = [
 const SUPPORTED_AUTHENTICATION_PROTOCOLS_TEXT =
   commaSeparatedValues(ALLOWED_AUTH_METHODS);
 
-exports.command = 'auth';
-exports.describe = i18n(`${i18nKey}.describe`, {
+export const command = 'auth';
+export const describe = i18n(`${i18nKey}.describe`, {
   configName: DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME,
 });
 
-exports.handler = async options => {
+type AuthArgs = CommonArgs &
+  ConfigArgs &
+  TestingArgs & {
+    authType?: string;
+    account?: string;
+  };
+
+export async function handler(
+  args: ArgumentsCamelCase<AuthArgs>
+): Promise<void> {
   const {
     authType: authTypeFlagValue,
     config: configFlagValue,
     qa,
     providedAccountId,
-  } = options;
+  } = args;
   const authType =
     (authTypeFlagValue && authTypeFlagValue.toLowerCase()) ||
     PERSONAL_ACCESS_KEY_AUTH_METHOD.value;
-  setLogLevel(options);
+  setLogLevel(args);
 
   const env = qa ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD;
   // Needed to load deprecated config
-  loadConfig(configFlagValue);
-  checkAndWarnGitInclusion(getConfigPath());
+  loadConfig(configFlagValue!);
+  const configPath = getConfigPath();
+  if (configPath) {
+    checkAndWarnGitInclusion(configPath);
+  }
 
   if (!getConfigPath(configFlagValue)) {
     logger.error(i18n(`${i18nKey}.errors.noConfigFileFound`));
     process.exit(EXIT_CODES.ERROR);
   }
 
   trackCommandUsage('auth');
-  trackAuthAction('auth', authType, TRACKING_STATUS.STARTED);
+  trackAuthAction(
+    'auth',
+    authType,
+    TRACKING_STATUS.STARTED,
+    providedAccountId!
```

Comment:
Not sure it makes sense to assert that `providedAccountId` exists since it doesn't unless the `--account` flag is used. It looks like we could update `trackAuthAction` to accept `undefined`

Reply:
Good call, done in [be71b49](https://github.com/HubSpot/hubspot-cli/pull/1401/commits/be71b495f9f24049001fda5d7c3328c7479fcdc8)